From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <yuki.utk8g@gmail.com>
Date: Sat, 4 Apr 2026 18:42:15 +0300
Subject: [PATCH] (Annoyances) Unregister url schemes

Useless. Portable Linux mode is not supported, and provided .desktop
file handles them just fine.

Type "registertg" in settings to clean leftover files.

diff --git a/Telegram/SourceFiles/core/application.cpp b/Telegram/SourceFiles/core/application.cpp
index 439e0aa009..1e0dbf6a9e 100644
--- a/Telegram/SourceFiles/core/application.cpp
+++ b/Telegram/SourceFiles/core/application.cpp
@@ -2036,7 +2036,7 @@ void Application::RegisterUrlScheme() {
 		? u"-workdir \"%1\""_q.arg(cWorkingDir())
 		: QString();
 
-	base::Platform::RegisterUrlScheme(base::Platform::UrlSchemeDescriptor{
+	base::Platform::UnregisterUrlScheme(base::Platform::UrlSchemeDescriptor{
 		.executable = Platform::ExecutablePathForShortcuts(),
 		.arguments = arguments,
 		.protocol = u"tg"_q,
@@ -2047,7 +2047,7 @@ void Application::RegisterUrlScheme() {
 		.displayAppDescription = AppName.utf16(),
 	});
 
-	base::Platform::RegisterUrlScheme(base::Platform::UrlSchemeDescriptor{
+	base::Platform::UnregisterUrlScheme(base::Platform::UrlSchemeDescriptor{
 		.executable = Platform::ExecutablePathForShortcuts(),
 		.arguments = arguments,
 		.protocol = u"tonsite"_q,
diff --git a/Telegram/SourceFiles/settings/settings_codes.cpp b/Telegram/SourceFiles/settings/settings_codes.cpp
index d38b9312d2..aaa22f5865 100644
--- a/Telegram/SourceFiles/settings/settings_codes.cpp
+++ b/Telegram/SourceFiles/settings/settings_codes.cpp
@@ -167,7 +167,7 @@ auto GenerateCodes() {
 	});
 	codes.emplace(u"registertg"_q, [](SessionController *window) {
 		Core::Application::RegisterUrlScheme();
-		Ui::Toast::Show("Forced custom scheme register.");
+		Ui::Toast::Show("Cleaned custom scheme register.");
 	});
 	codes.emplace(u"numberbuttons"_q, [](SessionController *window) {
 		using namespace base::options;
